scipy/special/_basic.py

```diff
@@ -2940,17 +3049,38 @@ def factorialk(n, k, exact=True):
     10
 
     """
-    if exact:
-        if n < 1-k:
+    if not np.issubdtype(type(k), np.integer) or k < 1:
+        raise ValueError(f"k must be a positive integer, received: {k}")
+    if not exact:
+        raise NotImplementedError
+
+    helpmsg = ""
+    if k in {1, 2}:
+        func = "factorial" if k == 1 else "factorial2"
+        helpmsg = f"\nYou can try to use {func} instead"
+
+    # don't use isscalar due to numpy/numpy#23574; 0-dim arrays treated below
+    if np.ndim(n) == 0 and not isinstance(n, np.ndarray):
+        # scalar cases
+        if n is None or np.isnan(n):
+            return np.nan
+        elif not np.issubdtype(type(n), np.integer):
+            msg = "factorialk does not support non-integral scalar arguments!"
+            raise ValueError(msg + helpmsg)
```

Comment:
This error isn't hit during testing

Reply:
Fixed in https://github.com/scipy/scipy/pull/15841/commits/5daaf00fa4e977ddd17045e81ca04e0ee868f815